jcache example with multiple clients doesn't work

When running `ClientDataProducer` and `ClientDataConsumer` application throws exception 

```
Exception in thread "main" javax.cache.CacheException: Error opening URI [file:/Users/vikgamov/projects/hazelcast-code-samples/jcache/src/main/resources/hazelcast-client-c1.xml]
	at com.hazelcast.cache.impl.AbstractHazelcastCachingProvider.getCacheManager(AbstractHazelcastCachingProvider.java:97)
	at com.hazelcast.cache.impl.AbstractHazelcastCachingProvider.getCacheManager(AbstractHazelcastCachingProvider.java:121)
	at com.hazelcast.cache.HazelcastCachingProvider.getCacheManager(HazelcastCachingProvider.java:151)
	at com.hazelcast.examples.AbstractApp.initCacheManager(AbstractApp.java:57)
	at com.hazelcast.examples.jcache.client.ClientDataProducer.runApp(ClientDataProducer.java:34)
	at com.hazelcast.examples.jcache.client.ClientDataProducer.main(ClientDataProducer.java:23)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:497)
	at com.intellij.rt.execution.application.AppMain.main(AppMain.java:144)
Caused by: java.lang.IllegalArgumentException: Not available Hazelcast instance. Please specify your Hazelcast configuration file path via "HazelcastCachingProvider.HAZELCAST_CONFIG_LOCATION" property or specify Hazelcast instance name via "HazelcastCachingProvider.HAZELCAST_INSTANCE_NAME" property in "properties" parameter.
	at com.hazelcast.client.cache.impl.HazelcastClientCachingProvider.createHazelcastCacheManager(HazelcastClientCachingProvider.java:62)
	at com.hazelcast.client.cache.impl.HazelcastClientCachingProvider.createHazelcastCacheManager(HazelcastClientCachingProvider.java:38)
	at com.hazelcast.cache.impl.AbstractHazelcastCachingProvider.getCacheManager(AbstractHazelcastCachingProvider.java:94)
	... 10 more
```

--- a/jcache/src/main/java/com/hazelcast/examples/AbstractApp.java
+++ b/jcache/src/main/java/com/hazelcast/examples/AbstractApp.java
@@ -10,6 +10,7 @@ import javax.cache.spi.CachingProvider;
 import java.io.File;
 import java.net.URI;
 import java.util.Iterator;
+import java.util.Properties;
 import java.util.Random;
 import java.util.concurrent.TimeUnit;
 
@@ -52,9 +53,13 @@ public class AbstractApp {
      * initialize the JCache Manager that we will use for creating and getting a cache object
      */
     public CacheManager initCacheManager(URI uri) {
+        return initCacheManager(uri, null);
+    }
+
+    public CacheManager initCacheManager(URI uri, Properties properties) {
         //resolve a cache manager
         cachingProvider = Caching.getCachingProvider();
-        return cachingProvider.getCacheManager(uri, null);
+        return cachingProvider.getCacheManager(uri, null, properties);
     }
 
     public CacheManager initCacheManager() {

--- a/jcache/src/main/java/com/hazelcast/examples/jcache/client/ClientDataConsumer.java
+++ b/jcache/src/main/java/com/hazelcast/examples/jcache/client/ClientDataConsumer.java
@@ -1,11 +1,14 @@
 package com.hazelcast.examples.jcache.client;
 
+import com.hazelcast.cache.HazelcastCachingProvider;
 import com.hazelcast.cache.ICache;
 import com.hazelcast.examples.AbstractApp;
 
 import javax.cache.Cache;
 import javax.cache.CacheManager;
+import java.net.URI;
 import java.util.Iterator;
+import java.util.Properties;
 
 /**
  * This will try to iterate over cache values and remove them
@@ -26,9 +29,12 @@ public class ClientDataConsumer extends AbstractApp{
         //Force client be used as a provider
         clientSetup();
 
+        Properties properties1 = HazelcastCachingProvider.propertiesByLocation("classpath:hazelcast-client-c1.xml");
+        Properties properties2 = HazelcastCachingProvider.propertiesByLocation("classpath:hazelcast-client-c2.xml");
+
         //first thin is we need to initialize the cache Managers for each cluster
-        final CacheManager cacheManager1 = initCacheManager(uri1);
-        final CacheManager cacheManager2 = initCacheManager(uri2);
+        final CacheManager cacheManager1 = initCacheManager(uri1, properties1);
+        final CacheManager cacheManager2 = initCacheManager(uri2, properties2);
 
         //create a cache with the provided name
         final Cache<String, Integer> cacheAtCluster1 = cacheManager1.getCache("theCache",String.class, Integer.class);
@@ -43,48 +49,54 @@ public class ClientDataConsumer extends AbstractApp{
     }
 
     public void startConsumerTask(final Cache<String, Integer> cacheAtCluster) {
-        Thread t = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                while (!Thread.interrupted()){
-                    final Iterator<Cache.Entry<String, Integer>> iterator = cacheAtCluster.iterator();
-                    while (iterator.hasNext()) {
-                        final Cache.Entry<String, Integer> entry = iterator.next();
-                        //value maybe null which means it is expired
-                        if(entry.getValue() != null) {
-                            //poor entry just remove
-                            iterator.remove();
-                            //do something with entry, save to db etc.
+        if (cacheAtCluster != null) {
+            Thread t = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    while (!Thread.interrupted()){
+                        final Iterator<Cache.Entry<String, Integer>> iterator = cacheAtCluster.iterator();
+                        while (iterator.hasNext()) {
+                            final Cache.Entry<String, Integer> entry = iterator.next();
+                            //value maybe null which means it is expired
+                            if(entry.getValue() != null) {
+                                //poor entry just remove
+                                iterator.remove();
+                                //do something with entry, save to db etc.
+                            }
+                        }
+                        try {
+                            Thread.sleep(1000);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                            return;
                         }
                     }
-                    try {
-                        Thread.sleep(1000);
-                    } catch (InterruptedException e) {
-                        e.printStackTrace();
-                        return;
-                    }
                 }
-            }
-        });
-        t.start();
+            });
+            t.start();
+        } else {
+            System.out.println("Provided cache doesn't exist on cluster");
+        }
     }
 
     private void displayCacheSize(final String alias, final ICache<String, Integer> icache) {
-        Thread t = new Thread(new Runnable() {
-            @Override
-            public void run() {
-                while (!Thread.interrupted()){
-                    System.out.println("Cache: "+alias+" size:"+icache.size());
-                    try {
-                        Thread.sleep(2000);
-                    } catch (InterruptedException e) {
-                        e.printStackTrace();
-                        return;
+        if (icache != null) {
+            Thread t = new Thread(new Runnable() {
+                @Override
+                public void run() {
+                    while (!Thread.interrupted()){
+                        System.out.println("Cache: "+alias+" size:"+icache.size());
+                        try {
+                            Thread.sleep(2000);
+                        } catch (InterruptedException e) {
+                            e.printStackTrace();
+                            return;
+                        }
                     }
                 }
-            }
-        });
-        t.start();
+            });
+            t.start();
+        }
     }
 
 }

--- a/jcache/src/main/java/com/hazelcast/examples/jcache/client/ClientDataProducer.java
+++ b/jcache/src/main/java/com/hazelcast/examples/jcache/client/ClientDataProducer.java
@@ -1,10 +1,12 @@
 package com.hazelcast.examples.jcache.client;
 
+import com.hazelcast.cache.HazelcastCachingProvider;
 import com.hazelcast.examples.AbstractApp;
 
 import javax.cache.Cache;
 import javax.cache.CacheManager;
 import javax.cache.expiry.Duration;
+import java.util.Properties;
 import java.util.Random;
 import java.util.UUID;
 import java.util.concurrent.TimeUnit;
@@ -30,9 +32,12 @@ public class ClientDataProducer extends AbstractApp{
         //Force client be used as a provider
         clientSetup();
 
+        Properties properties1 = HazelcastCachingProvider.propertiesByLocation("classpath:hazelcast-client-c1.xml");
+        Properties properties2 = HazelcastCachingProvider.propertiesByLocation("classpath:hazelcast-client-c2.xml");
+
         //first thin is we need to initialize the cache Managers for each cluster
-        final CacheManager cacheManager1 = initCacheManager(uri1);
-        final CacheManager cacheManager2 = initCacheManager(uri2);
+        final CacheManager cacheManager1 = initCacheManager(uri1, properties1);
+        final CacheManager cacheManager2 = initCacheManager(uri2, properties2);
 
         //create a cache with the provided name
         final Cache<String, Integer> cacheAtCluster1 = initCache("theCache", cacheManager1, new Duration(TimeUnit.SECONDS,10));

